src/prof_data.c

```diff
@@ -1492,13 +1491,11 @@ prof_tctx_destroy(tsd_t *tsd, prof_tctx_t *tctx) {
 	}
 	malloc_mutex_unlock(tsd_tsdn(tsd), gctx->lock);
 	if (destroy_gctx) {
-		prof_gctx_try_destroy(tsd, prof_tdata_get(tsd, false), gctx,
```

Comment:
Also I couldn’t find where this (line 1480) happens: 

> Change state such that the dumping thread will complete destruction during a late dump iteration phase.

Yinan can you please verify that’s still true?

Reply:
Yes, it's true. See `prof_gctx_finish()` - it deletes any `purgatory` `tctx` at the end of the dumping process and also deletes the `gctx` if needed.